alacritty/src/display/window.rs

```diff
@@ -433,11 +436,22 @@ impl Window {
         self.window.set_simple_fullscreen(simple_fullscreen);
     }
 
-    pub fn set_ime_allowed(&self, allowed: bool) {
+    /// Set IME inhibitor state.
+    pub fn set_ime_inhibitor(&mut self, inhibitor: ImeInhibitor, inhibit: bool) {
+        if self.ime_inhibitor.contains(inhibitor) != inhibit {
+            self.ime_inhibitor.set(inhibitor, inhibit);
+            self.update_ime_allowed();
+        }
+    }
+
+    /// Update whether IME should be offered.
+    fn update_ime_allowed(&mut self) {
```

Comment:
This can be inlined at this point and the docs must reflect IME update, etc in `set_ime_inhibitor`.